flocker/node/agents/ebs.py

```diff
@@ -107,6 +115,68 @@ def _wait_for_volume(expected_volume,
             )
 
 
+def _check_blockdevice_size(device, size):
+    """
+    Helper function to check if the size of block device with given
+    suffix matches the input size.
+
+    :param unicode device: Name of the block device to check for size.
+    :param int size: Size, in SI metric bytes, of device we are interested in.
+
+    :returns True if a block device with given name has given size.
+        False otherwise.
+    """
+    device_name = b"/dev/" + device
+    command = [b"lsblk", b"-nb", b"-o", b"SIZE", device_name]
+    device_size = int(check_output(command).strip().decode("ascii"))
+
+    # OS reports device size in powers of 2, where has EBS
+    # talks powers of 10. Hence, convert before comparing.
+    if size/(1000**3) == device_size/(1024**3):
```

Comment:
`size` and `device_size` are both in bytes.  I would have expected `size == device_size` to be the reasonable comparison to make here.

If `size/(1000**3) == device_size/(1024**3)` determines the sizes match, isn't it because either:

  * this uses integer division so many different sizes are effectively rounded to the same number of GB or GiB (eg, `3 * 10 ** 9 / 2 ** 30 == 3` even though there's a clear unit conversion error there)
  * or the Amazon volumes are being created with the wrong size which the unit mismatch here exactly masks (looking at http://boto.readthedocs.org/en/latest/ref/ec2.html#boto.ec2.connection.EC2Connection.create_volume I think this might be the case, it says "GiB" but I think the `create_volume` implementation passes in a value as if it were meant to be GB)